packages/web-components/fast-foundation/src/tree-item/tree-item.ts

```diff
@@ -14,7 +15,7 @@ import { applyMixins } from "../utilities/apply-mixins.js";
  * @remarks
  * determines if element is an HTMLElement and if it has the role treeitem
  */
-export function isTreeItemElement(el: Element): el is HTMLElement {
+export function isTreeItemElement(el: Element): el is FASTTreeItem {
```

Comment:
This was intentionally broad - if I slot my own element with a role of `treeitem` this should still work, which is why we are checking for `role` first.

Reply:
This is a type predicate - stating `el is HTMLElement` only tells the engine to treat `el` as if it's an `HTMLElement`: https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates

Reply:
Sorry, to be clear - my point is that FASTTreeItem is a more _narrow_ definition that what should actually be allowable here. An HTMLElement such as "div" with a role of `treeitem` or a non-fast tree item would not be supported here by way of the typing.